crates/viewer/re_space_view_map/src/map_space_view.rs

```diff
@@ -0,0 +1,334 @@
+                state,
+            );
+        });
+
+        // "Center and follow" button to reset view following mode after interacting
+        // with the map.
+        let map_state = state.downcast_mut::<MapSpaceViewState>()?;
+        ui.horizontal(|ui| {
+            let is_detached = map_state.map_memory.detached().is_none();
+
+            if !is_detached && ui.button("Center and follow positions").clicked() {
+                map_state.map_memory.follow_my_position();
+            }
+        });
+
+        Ok(())
+    }
+
+    fn ui(
+        &self,
+        ctx: &ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        state: &mut dyn SpaceViewState,
+
+        query: &ViewQuery<'_>,
+        system_output: SystemExecutionOutput,
+    ) -> Result<(), SpaceViewSystemExecutionError> {
+        let state = state.downcast_mut::<MapSpaceViewState>()?;
+        let map_viz_system = system_output.view_systems.get::<MapVisualizerSystem>()?;
+
+        let blueprint_db = ctx.blueprint_db();
+        let view_id = query.space_view_id;
+        let map_options =
+            ViewProperty::from_archetype::<MapOptions>(blueprint_db, ctx.blueprint_query, view_id);
+        let map_provider = map_options.component_or_fallback::<MapProvider>(ctx, self, state)?;
+        let zoom_level = map_options
+            .component_or_fallback::<ZoomLevel>(ctx, self, state)?
+            .0;
+        let access_token = map_options
+            .component_or_fallback::<Secret>(ctx, self, state)?
+            .0
+            .to_string();
+
+        if state.map_memory.set_zoom(*zoom_level).is_err() {
+            re_log::warn_once!(
+                "Failed to set zoom level for map. Zoom level should be between zero and 22"
+            );
+        };
+
+        // if state changed let's update it from the blueprint
+        if state.selected_provider != map_provider || access_token != state.access_token {
+            state.tiles = None;
+            state.selected_provider = map_provider;
+            state.access_token = access_token;
+        }
+
+        let (tiles, map_memory) = match state.ensure_and_get_mut_refs(ui.ctx()) {
+            Ok(refs) => refs,
+            Err(err) => return Err(err),
+        };
+
+        egui::Frame::default().show(ui, |ui| {
+            let some_tiles_manager: Option<&mut dyn Tiles> = Some(tiles);
+            let map_widget = ui.add(
+                Map::new(
+                    some_tiles_manager,
+                    map_memory,
+                    map_viz_system
+                        .map_entries
+                        .first()
+                        .unwrap_or(&MapEntry::default())
+                        .position,
+                )
+                .with_plugin(PositionsOnMap {
+                    positions: map_viz_system.map_entries.clone(),
+                }),
+            );
+
+            map_widget.double_clicked().then(|| {
+                map_memory.follow_my_position();
```

Comment:
Our usual semantics for double clicking is zoom to the "bounding box", e.g. whatever area that contains the entire view content.

I haven't looked into it yet. If it is involved, we can keep that for a follow-up PR though.

Reply:
double click reset the view (re-enable follow mode). We can change the zoom level as well, but if you have only point finding what zoom level the user expects is tricky imu